src/lib/reasoners/bitv.ml

```diff
@@ -729,51 +1154,96 @@ module Shostak(X : ALIEN) = struct
 
   let print = Debug.print_C_ast
 
+  let nat_of_bv term bv m =
```

Comment:
Is it clear that the fully-expanded version of `BV2Nat` is what we want rather than only converting fully interpreted bvs?

Reply:
It is not, I'll fix this.

Reply:
It will be done in a separate PR